core/Segment.php

```diff
@@ -110,6 +110,35 @@ public function __construct($segmentCondition, $idSites)
         }
     }
 
+    private function getAvailableSegments()
+    {
+        // segment metadata
+        if (empty($this->availableSegments)) {
+            $this->availableSegments = API::getInstance()->getSegmentsMetadata($this->idSites, $_hideImplementationData = false);
```

Comment:
Actually I might need similar feature somewhere else and might introduce a class for this. Will do it in another branch then as this one might be merged sooner. Problem is that `core/Segment` is a bit broken by design as it is in core but accesses the API. It should actually not access anything from a plugin... this would require lots of refactoring though I reckon